internal/services/compute/shared_image_gallery_resource.go

```diff
@@ -154,9 +232,120 @@ func resourceSharedImageGalleryDelete(d *pluginsdk.ResourceData, meta interface{
 		return err
 	}
 
+	resp, err := client.Get(ctx, *id, galleries.DefaultGetOperationOptions())
+	if err != nil {
+		return fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	if model := resp.Model; model != nil {
+		if prop := model.Properties; prop != nil && prop.SharingProfile != nil && prop.SharingProfile.Permissions != nil {
+			if *prop.SharingProfile.Permissions == galleries.GallerySharingPermissionTypesCommunity {
```

Comment:
```suggestion
			if pointer.From(prop.SharingProfile.Permissions) == galleries.GallerySharingPermissionTypesCommunity {
```
To be consistent in the code should we migrate this to use the `pointer` helper? 

NOTE: I am not going to mark all of the `*` code, but there are a lot of places in the code where this can be migrated as well. WDYT?